File: c++/namespaceTest.cpp
#include <iostream>

using namespace std;

namespace n1{
	int add(int a, int b){
		return a+b;
	}
}

namespace n2{
	int add(int a, int b){
		return 100 + a + b;
	}
}

void test(){
	using namespace n1;
	int c = add(10, 20);
	cout << c;
}

void test1(){
	int c = n1::add(10, 20);
	cout << c;
}

void test2(){
	using namespace n2;
	int c = add(10, 20);
	cout << c;
}

void test3(){
	int c = n2::add(10, 20);
	cout << c;
}

int main(){
	test3();
	return 0;
}

File: python/namespaceTest.py

a=1

def change():
	a=100

def change2():
	global a
	a=100

def change3():
	a=100
	def change():
		a=200
		print(a)
	change()
	print(a)

def change4():
	a=100
	def change():
		nonlocal a
		a=200
		print(a)
	change()
	print(a)

def test():
	change()
	print(a)

def test2():
	change2()
	print(a)

def test3():
	change3()
	print(a)

def test4():
	change4()
	print(a)

test4()
